internal/ociartifact/store.go

```diff
@@ -168,7 +168,13 @@ func (s *Store) PullManifest(
 		manifest.DockerV2Schema2ConfigMediaType,
 	}
 
-	if slices.Contains(imageMimeTypes, mimeType) && slices.Contains(configMediaTypes, mediaType) {
+	artifactType := ""
+	if ociManifest, err := manifest.OCI1FromManifest(manifestBytes); err == nil {
+		artifactType = ociManifest.ArtifactType
+		log.Debugf(ctx, "Found artifact type: %s", artifactType)
+	}
```

Comment:
nit: we can check the artifact type only when `slices.Contains(imageMimeTypes, mimeType) && slices.Contains(configMediaTypes, mediaType)` is met, then we can reduce unnecessary manifest parsing.
It will make the additional nest though...

Reply:
Yep, fixed as intended.